src/test/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorServiceTest.java

```diff
@@ -123,4 +124,52 @@
         }
     }
 
+    @Issue("JENKINS-58501")
+    @Ignore
+    @Test public void mismatchMetaProgrammingFalsePositives() throws Exception {
+        boolean origFailOnMismatch = CpsVmExecutorService.FAIL_ON_MISMATCH;
+        CpsVmExecutorService.FAIL_ON_MISMATCH = false;
+        try {
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+            errors.checkSucceeds(() -> {
+                p.setDefinition(new CpsFlowDefinition(
+                    "import org.codehaus.groovy.runtime.InvokerHelper \n" + 
+                    "c = { println 'doing a thing' } \n" +
+                    "InvokerHelper.getMetaClass(c).invokeMethod(c, 'call', null)", false));
+                WorkflowRun b = r.buildAndAssertSuccess(p);
+                r.assertLogNotContains("MetaClassImpl", b);
```

Comment:
nit: nice to do something like this to make it extra clear what is happening without even having to run the test: 

```suggestion
                r.assertLogNotContains(CpsVmExecutorService.mismatchMessage("MetaClassImpl", "invokeMethod", "CpsClosure2", "call"), b); // Or whatever the right values are.
```